apps/typegpu-docs/src/pages/index.astro

```diff
@@ -150,6 +158,87 @@ const showcaseVideoResolution = [2048, 1200];
       </div>
     </section>
 
+    {/* Roadmap Section */}
+    <div
+      class="box-border flex flex-col items-center bg-white w-full my-32 py-8 md:p-16 lg:p-32 md:rounded-[5rem] rounded-[2.5rem]">
+      <section
+        class="box-border flex flex-col items-center justify-center px-4 lg:gap-20 gap-10 md:gap-[3.75rem] lg:max-w-[85rem] md:max-w-[53.5rem] max-w-[34.5rem]">
+        <div>
+          <h2 class="mb-4 font-normal text-center md:mb-8 lg:mb-8 center">
+            Roadmap
+          </h2>
+          <p class="text-center text-pretty max-w-[60rem]">
+            The road to end-to-end type safety on the GPU involves implementing
+            many interoperating primitives. Below is our roadmap of features
+            that aim to fulfill that goal.
+          </p>
+        </div>
+
+        <div class="max-w-[50rem]">
+          <ol class="flex flex-col m-0 p-0">
+            <RoadMapStep title="Data structures & buffers" releasedIn="0.1">
+              <MilestoneBuffersBadge slot="visual" complete />
+              The WebGPU Shading Language (WGSL) gives developers a rich sense of
+              the structure of the data that it operates on. This is lost when accessed
+              through JavaScript, as the APIs accept raw byte buffers. TypeGPU regains
+              that information on the JavaScript side and handles proper byte alignment
+              and padding automatically.
+            </RoadMapStep>
+            <RoadMapStep title="Bind groups">
+              <MilestoneBindGroupsBadge slot="visual" />
+              Connecting resources to shaders via numeric indices can lead to problems
+              that are hard to catch at build-time/CI. By using named keys and typed
+              data structures we can ensure that the provided resources are what
+              the shader expects.
+            </RoadMapStep>
+            <RoadMapStep title="Functions">
+              <MilestoneFunctionsBadge slot="visual" />
+              By wrapping each function with a "typed shell", they can be moved across
+              different files/modules. Any resources (variables, buffers, etc.) in
+              the external scope can be accessed inside the function.
+            </RoadMapStep>
+            <RoadMapStep title="Pipelines">
+              <MilestonePipelinesBadge slot="visual" />
+              To ensure that shader stages match each-other, and pipeline constants
+              are filled with proper values, TypeGPU plans to provide typed pipelines.
+            </RoadMapStep>
+            <RoadMapStep title="Imperative code">
+              <MilestoneCodeBadge slot="visual" />
+              By giving developers the ability to write imperative shader code in
+              TypeScript, we can leverate the excellent TypeScript LSP to bring end-to-end
+              typesafety to shader code like never before.
```

Comment:
Shader code is type safe - the interaction with JS isn't. I would say we bring type safety to applications utilizing gpu, not the shader code.

Reply:
WDYM by interaction with JS?

Reply:
The statement is correct, we will bring type safety to shader code - but `.wgsl` files are already type safe. WebGPU API isn't type safe - I would say we extend the type safety to the API land (while maintaining it inside the shader code)

Reply:
Aah, I see.

> I would say we extend the type safety to the API land (while maintaining it inside the shader code)

In my mind, all the previous milestones are responsible for maximizing type-safety in API land, and this one finally connects the two "type safeties" together. 🤔

Reply:
> we can leverate the excellent TypeScript LSP to complete the type safety puzzle, bridging the gap between our API and type safe shader code.

Reply:
@reczkok Thoughts? 👀